### ISSUE TITLE
adding item to array for multiple paths with unexpected result

### ISSUE THREAD

**Issue by ak-at-gvl** [mikefarah/yq#750]
**Bug-Desciption**
My use case is to add an environment variable to all given services in a docker-compose.yml file – without knowing the service names.
Unfortunately the specific list of environment-variables of each services is not appended with the new variable, but only the _very first_ occurrence of the array is appended correctly and then copied wrongly the whole resulting array of the first service to all other services.

Version of yq: 4.4.1 & 4.6.1
Operating system: mac & linux
Installed on linux via: 
```
wget "https://github.com/mikefarah/yq/releases/download/v4.6.1/yq_linux_amd64" -O /usr/bin/yq && chmod +x /usr/bin/yq
```
on mac via homebrew

**Input Yaml**

sample.yaml:
```yaml
version: "1.2.3"
services:
  svc1:
    environment:
      - ENV_VAR1_1
      - ENV_VAR1_2
  svc2:
    environment:
      - ENV_VAR2_1
      - ENV_VAR2_2
      - ENV_VAR2_3
      - ENV_VAR2_4
  svc3:
    environment:
      - ENV_VAR3_1
      - ENV_VAR3_2
      - ENV_VAR3_3
  svc4:
    image: busybox:latest
```

**Command**
The command i ran:
```
yq eval '.services[].environment += ["NEW_ARRAY_ENTRY"]' sample.yaml
```

**Actual behavior**

```yaml
version: "1.2.3"
services:
  svc1:
    environment:
      - ENV_VAR1_1
      - ENV_VAR1_2
      - NEW_ARRAY_ENTRY
  svc2:
    environment:
      - ENV_VAR1_1
      - ENV_VAR1_2
      - NEW_ARRAY_ENTRY
  svc3:
    environment:
      - ENV_VAR1_1
      - ENV_VAR1_2
      - NEW_ARRAY_ENTRY
  svc4:
    image: busybox:latest
    environment:
      - ENV_VAR1_1
      - ENV_VAR1_2
      - NEW_ARRAY_ENTRY
```

**Expected behavior**

```yaml
version: "1.2.3"
services:
  svc1:
    environment:
      - ENV_VAR1_1
      - ENV_VAR1_2
      - NEW_ARRAY_ENTRY
  svc2:
    environment:
      - ENV_VAR2_1
      - ENV_VAR2_2
      - ENV_VAR2_3
      - ENV_VAR2_4
      - NEW_ARRAY_ENTRY
  svc3:
    environment:
      - ENV_VAR3_1
      - ENV_VAR3_2
      - ENV_VAR3_3
      - NEW_ARRAY_ENTRY
  svc4:
    image: busybox:latest
    environment:
      - NEW_ARRAY_ENTRY
```

**Additional context**
My current workaround is to use `yq` to transform the file into JSON, modifying with `jq` and using `yq` again to re-transform the result into YAML – which is actually very ugly:
```
yq --tojson eval '.' docker-compose.yml > docker-compose.json && \
  jq '.services[].environment += ["NEW_ARRAY_ENTRY"]' docker-compose.json | \
  yq eval -P > docker-compose.yml && \
  rm docker-compose.json;
```

**Comment by mikefarah** [comment]
Looks like a bug with '+=' :( I'll look into it asap - as a work around you can:

```
yq e '.services.*.environment |= . + ["cat"]' examples/data1.yaml
```

**Comment by mikefarah** [comment]
Fixed in 4.6.2

**Comment by ak-at-gvl** [comment]
Great! Thanks a lot for fixing this quickly!
